src/jupyterlabcontroller/storage/kubernetes/creator.py

```diff
@@ -0,0 +1,157 @@
+"""Generic Kubernetes object storage supporting only create and read.
+
+Provides a generic Kubernetes object management class and instantiations of
+that class for Kubernetes object types that only support create and read.
+This is sufficient for a lot of object types the lab controller manipulates.
+Storage classes for object types that only need those operations are provided
+here.
+
+For object types that need to support other operations, see
+`KubernetesObjectDeleter`, which subclasses `KubernetesObjectCreator` and adds
+list and delete support, and its subclasses.
+"""
+
+from __future__ import annotations
+
+from collections.abc import Awaitable, Callable
+from typing import Any, Generic, TypeVar
+
+from kubernetes_asyncio import client
+from kubernetes_asyncio.client import ApiClient, ApiException, V1ConfigMap
+from structlog.stdlib import BoundLogger
+
+from ...exceptions import KubernetesError
+from ...models.domain.kubernetes import KubernetesModel
+
+#: Type of Kubernetes object being manipulated.
+T = TypeVar("T", bound=KubernetesModel)
+
+__all__ = [
+    "ConfigMapStorage",
+    "KubernetesObjectCreator",
+    "T",
+]
+
+
+class KubernetesObjectCreator(Generic[T]):
```

Comment:
We also deal with Volumes, which are not namespaced, so I feel like you may want to jam "Namespaced" into the class name, or have "NonNamespaced" variants of these for Volumes and any other non-namespaced objects we deal with.

Reply:
This is essentially the same problem we dealt with way back when with the Kubespawner Reflectors.